### DISCUSSION TITLE
Minestom does not stop sometimes

### DISCUSSION THREAD

**Discussion by offby0point5** [Minestom/Minestom#668] [Help] [answered]
I have a [server implementation (branch fix-restart)](https://github.com/KlainStom/microstom/tree/fix-restart) with a command to restart the server and an [extension](https://github.com/KlainStom/terminus) to allow executing commands over SSH.
Starting the server with a console works fine. After a restart it runs in background, so then I use the extension to access the console. When the server is restarted a second time it still works. But the third restart does not work, as the server does not fully stop. 
The log file of the last restart looks like: 
> Stopping Minestom server.
> Unloading all extensions.
> Unloading extension terminus
> terminus terminate.
> Close Shell for offbyone

while the working restart looks like:
> Stopping Minestom server.
> Unloading all extensions.
> Unloading extension terminus
> terminus terminate.
> Close Shell for offbyone
> Shutting down all thread pools.
> Minestom server stopped successfully.

I don't have any idea, why this is happening...

**Comment by TheMode** [comment]
This should likely be an issue instead of discussion.

I'd suggest first doing a thread dump to understand why the process is not stopped.

**Comment by offby0point5** [comment, marked as answer]
Not Minestom, but the SSH server extension is the problem. Somehow shutting down Apache SSHD hangs forever. Stopping the program wothout stopping the SSH server works fine.
